cli/src/commands/analyze.rs

```diff
@@ -0,0 +1,140 @@
+    }
+}
+
+/// Enumerates which SBOM analysis provider to employ.
+#[derive(Clone, Debug)]
+pub(crate) enum AnalysisProviderKind {
+    /// Generate a Detailed report from SBOM data.
+    Detail,
+}
+
+impl ValueEnum for AnalysisProviderKind {
+    fn value_variants<'a>() -> &'a [Self] {
+        &[Self::Detail]
+    }
+
+    fn to_possible_value(&self) -> Option<PossibleValue> {
+        Some(match self {
+            AnalysisProviderKind::Detail => PossibleValue::new("detail").help(
+                "Use the Detail Analytic to create comprehensive reports for all Primary SBOMs",
+            )
+        })
+    }
+}
+
+/// Specifies the CLI args for the SBOM command.
+#[derive(Debug, Parser)]
+pub struct AnalyzeArgs {
+
+    /// Specifies to run the command against the local debug environment.
+    #[arg(long)]
+    debug: bool,
+
+    /// Specifies the type of provider
+    provider: AnalysisProviderKind,
+
+    /// Flattened args for use with the Detail Provider
+    #[command(flatten)]
+    pub detail_args: Option<DetailArgs>,
+}
+
+/// Args for generating a Detailed Report
+#[derive(Clone, Debug, Parser)]
+pub struct DetailArgs {}
+
+/// Strategy pattern implementation that handles Report generation commands.
+struct DetailProvider {}
+
+impl DetailProvider {
+
+    async fn new_provider(
+        cx: Context,
+        storage: Arc<dyn StorageProvider>,
+    ) -> Result<AnalyticExecutionTask, Error> {
+
+        let store = Arc::new(
+            Store::new(&cx)
+                .await
+                .map_err(
+                    |e| Error::Sbom(e.to_string())
+                )?,
+        );
+
+        let provider = AnalyticExecutionTask::new(
+            AnalyticService::new(store.clone(), storage),
+        );
+
+        Ok(provider)
+    }
+
+    async fn execute(args: &AnalyzeArgs) -> Result<(), Error> {
+
+        let storage: Arc<dyn StorageProvider>;
+
+        let cx = match &args.debug {
+            false => {
+                storage = Arc::new(S3StorageProvider {});
+                harbcore::config::harbor_context().map_err(
+                    |e| Error::Config(e.to_string())
+                )?
+            }
+            true => {
+                storage = Arc::new(
+                    FileSystemStorageProvider::new(
+                        "/tmp/harbor-debug/detail".to_string(),
+                    )
+                );
+                harbcore::config::dev_context(None).map_err(
+                    |e| Error::Config(e.to_string())
+                )?
+            }
+        };
+
+        match &args.detail_args {
+            None => {
+
+                let mut task: Task = Task::new(
+                    TaskKind::Analytics(
+                        AnalyticProviderKind::Detail
+                    )
+                ).map_err(
+                    |e| Error::Enrich(e.to_string())
+                )?;
+
+                let mut provider: AnalyticExecutionTask
```

Comment:
Because I'm borrowing it when I invoke `provider.execute()` below and the compiler says: `Cannot borrow immutable local variable `provider` as mutable` if it's not mutable.

Reply:
Wrapped this up!